src/mlpack/core/data/matrix_options.hpp

```diff
@@ -112,6 +156,25 @@ class MatrixOptionsBase : public DataOptionsBase<MatrixOptionsBase<Derived>>
     return *this;
   }
 
+  // Augment with the options of the other `MatrixOptionsBase`.
+  template<typename Derived2>
+  void Combine(const MatrixOptionsBase<Derived2>& other)
+  {
+    // Combine the noTranspose option.
+    noTranspose =
+        DataOptionsBase<MatrixOptionsBase<Derived>>::CombineBooleanOption(
+        noTranspose, other.noTranspose, "NoTranspose()");
+
+    // If the derived type is the same, we can take any options from it.
+    if (std::is_same_v<Derived, Derived2>)
+    {
+      static_cast<Derived&>(*this).Combine(static_cast<const Derived&>(other));
```

Comment:
```suggestion
    if constexpr (std::is_same_v<Derived, Derived2>)
    {
      static_cast<Derived&>(*this).Combine(static_cast<const Derived2&>(other));
```

This instance needs constexpr too so that the compiler doesn't try and compile it when `Derived` is not equal to `Derived2`.  Up to you whether you want to change the `Derived` to `Derived2` in the `static_cast`, that really doesn't make a difference either way.